### ISSUE TITLE
Cannot exclude multiple groups

### ISSUE THREAD

**Issue by amimas** [jk1/Gradle-License-Report#88]
I cannot seem to be able to exclude more than one group using the `excludeGroups` property. I get a stackoverflow error message (see below).

This is what my configuration look like. I can specify each groups by themselves and the generated report does exclude that group, but I can't seem to specify both groups to be excluded.
```gradle
licenseReport {
	excludeGroups = ['org.amimas.project1', 'org.amimas.project2']
}
```

Stacktrace of the error:
```
* What went wrong:
Execution failed for task ':generateLicenseReport'.
> java.lang.StackOverflowError (no error message)

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':generateLicenseReport'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:596)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.StackOverflowError
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader.collectDependencies(ConfigurationReader.groovy:70)
        at com.github.jk1.license.reader.ConfigurationReader$_collectDependencies_closure1.doCall(ConfigurationReader.groovy:70)

```

**Comment by jk1** [comment]
Multiple groups exclude works for me in my own builds.
Can't there be a circular dependency between `org.amimas.project1` and `org.amimas.project2` ?

**Comment by amimas** [comment]
Hmm... I do have circular dependency between those 2 projects. Do you think that's the reason why these 2 groups couldn't be used together in the `excludeGroups` configuration?

**Comment by jk1** [comment]
Yes, I do. And can reproduce the bug with a dependency cycle involved.

**Comment by jk1** [comment]
Released in the scope of 1.2
